Failed compilation with --disable-boost-program-options

Modules environment:

<pre>
[05:28:29][pbauman@fry:/fry1/data/shared/software/builddir/queso-R5Kar7/queso-0.56.1]$ module list

Currently Loaded Modules:
  1) emacs/24.4   2) autotools/1.0   3) git/2.3.0   4) valgrind/3.11.0   5) paraview/4.3.1   6) cmake/3.5.2   7) texlive/2014   8) doxygen/1.8.9.1   9) gcc/6.3.0  10) mpich/3.2  11) gsl/2.1  12) glpk/4.60  13) hdf5/1.8.18

[05:29:29][pbauman@fry:/fry1/data/shared/software/builddir/queso-R5Kar7/queso-0.56.1]$
</pre>

Result of configure:
<pre>
----------------------------------- SUMMARY -----------------------------------

Package version............... : queso-0.56.1

Debug mode.................... : no
C++ compiler.................. : mpic++
C++ compiler flags............ : -g -O2 -Wall
C Preprocessor flags.......... : -DDISABLE_BOOST_PROGRAM_OPTIONS
 
GSL_LIBS...................... : -L/fry1/data/shared/software/libs/gsl/2.1/gcc/6.3.0/lib -lgsl -lgslcblas -lm
BOOST_DIR..................... :
 
Install dir................... : /usr/local
Build user.................... : pbauman
Build host.................... : fry.eng.buffalo.edu
Configure date................ : 2017-01-13 17:28
Build architecture............ : x86_64-unknown-linux-gnu
git revision.................. : N/A

Optional Features:
   Link with MPI.............. : yes
   Link with GRVY............. : no
   Link with GLPK............. : yes
   Link with HDF5............. : no
   Link with Trilinos......... : no
   Link with libmesh.......... : no
   Enable gcov code coverage.. : no
   Build internal ANN library. : no

Optional Feature Paths:
   GLPK DIR................... : /fry1/data/shared/software/libs/glpk/4.60/gcc/6.3.0

-------------------------------------------------------------------------------
</pre>


Compilation errors:
<pre>
  CXX      core/src/EnvironmentOptions.lo
core/src/EnvironmentOptions.C: In constructor ‘QUESO::EnvOptionsValues::EnvOptionsValues(const QUESO::BaseEnvironment*, const char*)’:
core/src/EnvironmentOptions.C:149:53: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, const char [1])’
   m_help = m_env->input()(m_option_help, UQ_ENV_HELP);
                                                     ^
core/src/EnvironmentOptions.C:150:101: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_numSubEnvironments = m_env->input()(m_option_numSubEnvironments, UQ_ENV_NUM_SUB_ENVIRONMENTS_ODV);
                                                                                                     ^
core/src/EnvironmentOptions.C:151:102: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, const char [2])’
   m_subDisplayFileName = m_env->input()(m_option_subDisplayFileName, UQ_ENV_SUB_DISPLAY_FILE_NAME_ODV);
                                                                                                      ^
core/src/EnvironmentOptions.C:152:102: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_subDisplayAllowAll = m_env->input()(m_option_subDisplayAllowAll, UQ_ENV_SUB_DISPLAY_ALLOW_ALL_ODV);
                                                                                                      ^
core/src/EnvironmentOptions.C:153:111: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_subDisplayAllowInter0 = m_env->input()(m_option_subDisplayAllowInter0, UQ_ENV_SUB_DISPLAY_ALLOW_INTER0_ODV);
                                                                                                               ^
core/src/EnvironmentOptions.C:156:37: error: invalid use of incomplete type ‘const class QUESO::GetPot’
   unsigned int size = m_env->input().vector_variable_size(m_option_subDisplayAllowedSet);
                                     ^
In file included from ../inc/queso/Miscellaneous.h:28:0,
                 from core/src/EnvironmentOptions.C:30:
../inc/queso/Environment.h:55:7: note: forward declaration of ‘class QUESO::GetPot’
 class GetPot;
       ^~~~~~
core/src/EnvironmentOptions.C:160:78: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, unsigned int&, unsigned int&)’
     unsigned int allowed = m_env->input()(m_option_subDisplayAllowedSet, i, i);
                                                                              ^
core/src/EnvironmentOptions.C:164:94: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_displayVerbosity = m_env->input()(m_option_displayVerbosity, UQ_ENV_DISPLAY_VERBOSITY_ODV);
                                                                                              ^
core/src/EnvironmentOptions.C:165:85: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_syncVerbosity = m_env->input()(m_option_syncVerbosity, UQ_ENV_SYNC_VERBOSITY_ODV);
                                                                                     ^
core/src/EnvironmentOptions.C:166:85: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_checkingLevel = m_env->input()(m_option_checkingLevel, UQ_ENV_CHECKING_LEVEL_ODV);
                                                                                     ^
core/src/EnvironmentOptions.C:167:67: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, const char [4])’
   m_rngType = m_env->input()(m_option_rngType, UQ_ENV_RNG_TYPE_ODV);
                                                                   ^
core/src/EnvironmentOptions.C:168:57: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, int)’
   m_seed = m_env->input()(m_option_seed, UQ_ENV_SEED_ODV);
                                                         ^
core/src/EnvironmentOptions.C:169:82: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, const char [1])’
   m_platformName = m_env->input()(m_option_platformName, UQ_ENV_PLATFORM_NAME_ODV);
                                                                                  ^
core/src/EnvironmentOptions.C:170:97: error: no match for call to ‘(const QUESO::GetPot) (std::__cxx11::string&, const char [1])’
   m_identifyingString = m_env->input()(m_option_identifyingString, UQ_ENV_IDENTIFYING_STRING_ODV);
                                                                                                 ^
core/src/EnvironmentOptions.C: In function ‘std::ostream& QUESO::operator<<(std::ostream&, const QUESO::EnvOptionsValues&)’:
core/src/EnvironmentOptions.C:242:21: error: invalid use of incomplete type ‘const class QUESO::GetPot’
   obj.m_env->input().print(os);
                     ^
In file included from ../inc/queso/Miscellaneous.h:28:0,
                 from core/src/EnvironmentOptions.C:30:
../inc/queso/Environment.h:55:7: note: forward declaration of ‘class QUESO::GetPot’
 class GetPot;
       ^~~~~~
make[3]: *** [core/src/EnvironmentOptions.lo] Error 1
</pre>

Sorry I don't have time to fix this myself at the moment (working on features instead), but I at least wanted to log it.

--- a/src/core/inc/Defines.h
+++ b/src/core/inc/Defines.h
@@ -54,6 +54,9 @@
 // Use GSL inline functions
 #define HAVE_INLINE
 
+// So we don't clash with other getpots
+#define GETPOT_NAMESPACE QUESO
+
 // And only do GSL range-checking if we're really debugging
 #ifndef DEBUG
 #define GSL_RANGE_CHECK_OFF

--- a/src/core/inc/InfiniteDimensionalMCMCSamplerOptions.h
+++ b/src/core/inc/InfiniteDimensionalMCMCSamplerOptions.h
@@ -28,6 +28,8 @@
 
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <queso/BoostInputOptionsParser.h>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 namespace QUESO {

--- a/src/core/src/Environment.C
+++ b/src/core/src/Environment.C
@@ -22,15 +22,15 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Environment.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
-#define GETPOT_NAMESPACE QUESO
 #include <queso/getpot.h>
 
 #include <queso/config_queso.h>
-#include <queso/Environment.h>
 #include <queso/EnvironmentOptions.h>
 #include <queso/RngGsl.h>
 #include <queso/RngBoost.h>

--- a/src/core/src/EnvironmentOptions.C
+++ b/src/core/src/EnvironmentOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/EnvironmentOptions.h>

--- a/src/core/src/OptimizerOptions.C
+++ b/src/core/src/OptimizerOptions.C
@@ -24,7 +24,13 @@
 
 #include <queso/Defines.h>
 #include <queso/Environment.h>
+
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <queso/BoostInputOptionsParser.h>
+#else
+#include <queso/getpot.h>
+#endif
+
 #include <queso/OptimizerOptions.h>
 
 namespace QUESO {

--- a/src/gp/src/ExperimentModelOptions.C
+++ b/src/gp/src/ExperimentModelOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/ExperimentModelOptions.h>

--- a/src/gp/src/GPMSAOptions.C
+++ b/src/gp/src/GPMSAOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/GPMSAOptions.h>
@@ -56,9 +60,9 @@ GPMSAOptions::GPMSAOptions(
 
 GPMSAOptions::GPMSAOptions()
   :
-  m_env(NULL),
+  m_env(NULL)
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
-  m_parser(new BoostInputOptionsParser())
+  ,m_parser(new BoostInputOptionsParser())
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 {
   this->set_defaults();

--- a/src/gp/src/GpmsaComputerModelOptions.C
+++ b/src/gp/src/GpmsaComputerModelOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/GpmsaComputerModelOptions.h>

--- a/src/gp/src/SimulationModelOptions.C
+++ b/src/gp/src/SimulationModelOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/SimulationModelOptions.h>

--- a/src/stats/inc/MLSamplingLevelOptions.h
+++ b/src/stats/inc/MLSamplingLevelOptions.h
@@ -29,7 +29,13 @@
 
 #include <queso/Environment.h>
 #include <queso/SequenceStatisticalOptions.h>
+
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <queso/BoostInputOptionsParser.h>
+#else
+#include <queso/getpot.h>
+#endif
+
 #define UQ_ML_SAMPLING_L_FILENAME_FOR_NO_FILE "."
 
 // _ODV = option default value

--- a/src/stats/src/MetropolisHastingsSGOptions.C
+++ b/src/stats/src/MetropolisHastingsSGOptions.C
@@ -22,11 +22,14 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Environment.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
-#include <queso/Environment.h>
 #include <queso/MetropolisHastingsSGOptions.h>
 #include <queso/Miscellaneous.h>
 

--- a/src/stats/src/MonteCarloSGOptions.C
+++ b/src/stats/src/MonteCarloSGOptions.C
@@ -22,8 +22,12 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
 #include <queso/MonteCarloSGOptions.h>

--- a/src/stats/src/StatisticalForwardProblemOptions.C
+++ b/src/stats/src/StatisticalForwardProblemOptions.C
@@ -22,11 +22,14 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif
 
-#include <queso/Defines.h>
 #include <queso/StatisticalForwardProblemOptions.h>
 #include <queso/Miscellaneous.h>
 

--- a/src/stats/src/StatisticalInverseProblemOptions.C
+++ b/src/stats/src/StatisticalInverseProblemOptions.C
@@ -22,11 +22,14 @@
 //
 //-----------------------------------------------------------------------el-
 
+#include <queso/Defines.h>
+
 #ifndef DISABLE_BOOST_PROGRAM_OPTIONS
 #include <boost/program_options.hpp>
+#else
+#include <queso/getpot.h>
 #endif  // DISABLE_BOOST_PROGRAM_OPTIONS
 
-#include <queso/Defines.h>
 #include <queso/StatisticalInverseProblemOptions.h>
 #include <queso/Miscellaneous.h>
 

